lib/winappdriver.js

```diff
@@ -103,28 +100,34 @@ class WinAppDriver extends events.EventEmitter {
   }
 
   async waitForOnline () {
-    // TODO WAD doesn't support the status command correctly, so just return
-    // true for now
-    return true;
 
     // we need to make sure WAD hasn't crashed
-    /*
     let winappdriverStopped = false;
     await retryInterval(20, 200, async () => {
       if (this.state === WinAppDriver.STATE_STOPPED) {
         // we are either stopped or stopping, so something went wrong
         winappdriverStopped = true;
         return;
       }
+      
+      else if (this.state === WinAppDriver.STATE_ONLINE) {
+        return;  
+      }
       await this.getStatus();
     });
+
     if (winappdriverStopped) {
       throw new Error('WinAppDriver crashed during startup.');
-    }*/
+    }
   }
 
   async getStatus () {
-    return await this.jwproxy.command('/status', 'GET');
+    var resBlock = await this.jwproxy.proxy('/status', 'GET');
+    if (resBlock[0].statusCode === 200){
+        log.info(`Status call returned 200`);
+        this.changeState(WinAppDriver.STATE_ONLINE);
```

Comment:
perhaps I don't know about the driver architecture, but this looks like a side effect to me if one changes the status in a method, which is just supposed to return a result. I'd rather rename it to `syncStatus`

Reply:
getStatus is a basedriver method so we should keep the name.  We can make the caller's responsibility to update the internal state.  I will change it to that as I'm fine either way.